.github/workflows/java-ec2-asg-e2e-test.yml

```diff
@@ -155,12 +155,33 @@ jobs:
             fi
           done
 
+      - name: Wait for ASG instance to be ready
+        run: |
+          # Wait for up to 15 minutes
+          retry_counter=0
+          max_retry=90
+          while [ $retry_counter -lt $max_retry ]; do
+            instance_id=$(aws autoscaling describe-auto-scaling-groups --auto-scaling-group-names ec2-single-asg-${{ env.TESTING_ID }} --region ${{ env.E2E_TEST_AWS_REGION }} --query "AutoScalingGroups[].Instances[0].InstanceId" --output text)
+            status_ok=$(aws ec2 describe-instance-status --instance-ids $instance_id | jq '(.InstanceStatuses[] | select(.InstanceStatus.Status == "ok" and .SystemStatus.Status == "ok")) != {}')
```

Comment:
Can we get throttled here? Calling this EC2 and ASG API every 10 seconds seems reasonable, just curious. For safety we can do 45 x 20s instead of 90 x 10s if you find that there's a concern, for example.

Reply:
Checked briefly but not a concern: https://docs.aws.amazon.com/ec2/latest/devguide/ec2-api-throttling.html